src/uuid/builder/uuidutils.rs

```diff
@@ -156,3 +165,34 @@ impl UuidUtils {
         }
     }
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Version {
+    /// An unknown version.
+    VersionUnknown = 0,
+    /// The randomly or pseudo-randomly generated version specified in RFC-4122.
+    VersionRandomBased = 4,
+    /// The time-ordered version with gregorian epoch proposed by Peabody and Davis.
+    VersionTimeOrdered = 6,
+    /// The custom or free-form version proposed by Peabody and Davis.
+    VersionUprotocol = 8,
+}
+
+impl Version {
+    /// Get the `Version` from the passed integer representation of the version.
+    /// Returns `None` if the value is not a valid version.
+    pub fn from_value(value: usize) -> Option<Self> {
+        match value {
+            0 => Some(Version::VersionUnknown),
+            4 => Some(Version::VersionRandomBased),
+            6 => Some(Version::VersionTimeOrdered),
+            8 => Some(Version::VersionUprotocol),
+            _ => None,
+        }
+    }
+
+    /// Returns the integer representation of the version.
```

Comment:
You can ensure that enum variants are represented as `usize` by prepending `#[repr(usize)]` to enum definition ([see doc](https://doc.rust-lang.org/nomicon/other-reprs.html#repru-repri)).
If you do so, everyone can safely cast the enum to usize.

Reply:
done